runtime/sema/gen/main.go

```diff
@@ -1583,12 +1584,14 @@ func simpleTypeLiteral(ty *typeDecl) dst.Expr {
 	//}
 
 	isResource := ty.compositeKind == common.CompositeKindResource
+	_, isPrimitive := sema.GeneratedPrimitiveSimpleTypes[ty.typeName]
```

Comment:
Similar to how other "kind"s are indicated through conformances in declarations of built-in types which produce Go code, like if a type is storable via conforming to the `Storable` type (also, `Equatable`, `Comparable`, etc), we might consider introducing a `Primitive` type in the code generator (which later then also could be introduced "properly" in the type checker.

Reply:
Done in https://github.com/onflow/cadence/pull/2962/commits/7b3805cd5b839f47c96336c13251b3d6b10cdd7b.